yarn-project/ethereum/src/l1_tx_utils.ts

```diff
@@ -306,15 +312,16 @@ export class L1TxUtils {
             try {
               const receipt = await this.publicClient.getTransactionReceipt({ hash });
               if (receipt) {
-                this.logger?.debug(`L1 transaction ${hash} mined`);
                 if (receipt.status === 'reverted') {
-                  this.logger?.error(`L1 transaction ${hash} reverted`);
+                  this.logger?.error(`L1 transaction ${hash} reverted`, receipt);
+                } else {
+                  this.logger?.debug(`L1 transaction ${hash} mined`);
                 }
                 return receipt;
               }
             } catch (err) {
               if (err instanceof Error && err.message.includes('reverted')) {
-                throw err;
+                throw formatViemError(err);
```

Comment:
What happens if we pass something that is not a `ViemError` to `formatViemError`? I'm worried we may get a non-viem error but still something that contains the string "reverted". But if we're fine sending anything into `formatViemError`, no worries then!

Reply:
yeah probably a good idea to check inside `formatViemError` and add a test case for it 👍